src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReaderState.cs

```diff
@@ -0,0 +1,133 @@
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item denotes the start of an indefinite-length UTF-8 text string (major type 3).
+        /// </summary>
+        StartTextString,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> is at the end of an indefinite-length UTF-8 text string (major type 3).
+        /// </summary>
+        EndTextString,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item denotes the start of an array (major type 4).
+        /// </summary>
+        StartArray,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> is at the end of an array (major type 4).
+        /// </summary>
+        EndArray,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item denotes the start of a map (major type 5).
+        /// </summary>
+        StartMap,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> is at the end of a map (major type 5).
+        /// </summary>
+        EndMap,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a semantic tag (major type 6).
+        /// </summary>
+        Tag,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a simple value (major type 7).
+        /// </summary>
+        SimpleValue,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is an IEEE 754 Half-Precision float (major type 7).
+        /// </summary>
+        HalfPrecisionFloat,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is an IEEE 754 Single-Precision float (major type 7).
+        /// </summary>
+        SinglePrecisionFloat,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is an IEEE 754 Double-Precision float (major type 7).
+        /// </summary>
+        DoublePrecisionFloat,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a <c>null</c> literal (major type 7).
+        /// </summary>
+        Null,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data items encodes a <see cref="bool" /> value (major type 7).
+        /// </summary>
+        Boolean,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> has completed reading a full CBOR document.
+        /// </summary>
+        /// <remarks>
+        ///   If <see cref="CborReader.AllowMultipleRootLevelValues"/> is set to <c>true</c>,
+        ///   the reader will report this value even if the buffer contains trailing bytes.
+        /// </remarks>
+        Finished,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> has encountered an incomplete CBOR document.
+        /// </summary>
+        EndOfData,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> has encountered an error in the CBOR format encoding.
```

Comment:
Ideally the modes wouldn't need to describe a type, but rather a concept.

```suggestion
        ///   Indicates that a reader has encountered an error in the CBOR format encoding.
```